### ISSUE TITLE
Relax editing metadata on published/posted materials

### ISSUE THREAD

**Issue by asmecher** [pkp/pkp-lib#10263]
**Describe the bug**
Currently, when content is published/posted in OJS/OMP/OPS, editors are restricted from making further changes:
![image](https://github.com/user-attachments/assets/772c3351-afd1-47a3-bfab-fb41c7f8458b)

In practice, however, editors want the ability to tweak/edit metadata in published content. This has led to anti-patterns like unpublish/edit/republish cycles.

In OMP, there is an additional problem, where certain data on Publication Formats can't be accessed/reviewed because the edit links disappear once the content is published.

Accepting that editors are going to want to tweak metadata after publication, we should keep the warning message present, but stop restricting edits. We should trust the editors to understand the implications of making edits, but not try to prevent it.

**Comment by ewhanson** [comment]
We had previously relied on the act of publishing (e.g. `Repo::publication()->publish()`) to indicate that metadata (new or changed) is available for downstream service to consume and used this as a jumping off point. By making it possible to edit this metadata post publication, I propose that we should treat each save event of the metadata as a `MetadataChanged` event and make use of the [MetadataChanged](https://github.com/pkp/pkp-lib/blob/main/classes/observers/events/MetadataChanged.php) and [BatchMetadataChanged](https://github.com/pkp/pkp-lib/blob/main/classes/observers/events/BatchMetadataChanged.php) events. 

I would propose a naive approach that simply always dispatches the `MetadataChanged` event when the form is saved/updated post publication, and let each individual implementation be in charge of deciding what metadata is important to process.

This approach has the advantage of using already existing events and jobs and allows any updates to happen asynchronously to the web request. It is also extensible for any future services or third-party plugins that need to know when metadata has been updated.

Examples of downstream services that would make use of this (either currently implemented or future implementations):

- Mark DOIs for a submission as stale. The DOI logic already handles resubmitting stale DOIs. This should already implemented for major the registration agencies we support.
- Update ORCID records for authenticated users in their ORCID profile. I don't think this has been implemented yet, but it could be added independently of this work.

**Comment by Hafsa-Naeem** [comment]
@asmecher Ready for review
**OJS_main:**
**ojs**: [pkp/pkp-lib#10263 Relax editing metadata on published/posted materials #4410](https://github.com/pkp/ojs/pull/4410)
**pkp/pkp-lib**: [pkp/pkp-lib#10263 edit metadata post publication #10335](https://github.com/pkp/pkp-lib/pull/10335)
**pkp/ui-library**: [pkp/pkp-lib#10263 Relax editing metadata on published/posted materials #398](https://github.com/pkp/ui-library/pull/398)

**Comment by asmecher** [comment]
Thanks, @Hafsa-Naeem! @Vitaliy-1, would you be able to review this one?

**Comment by jardakotesovec** [comment]
@asmecher @Hafsa-Naeem On ui-library side of things this make sense to apply to the new side modal workflow page. Trying to get that side modal workflow page as default on main branch ASAP..  Its probably 2-4 weeks away.

**Comment by ajnyga** [comment]
This is a good change, because journals get stuck and create new versions easily without any good reason (correcting a typo etc.)

Just asking that when we now edit the metadata, is this visible in the editorial history? ie. do we stamp who did what? I think this is important especially after something has been published.

**Comment by asmecher** [comment]
There would be an event in the event log, but if the editor chooses not to create a new version, the change will not be tracked.

Note that we'll be extending versioning further with https://github.com/pkp/pkp-lib/issues/4860 -- description there.

**Comment by jardakotesovec** [comment]
@Hafsa-Naeem Hi Hafsa, 
I am still active on this codebase, but risk of code conflicts is lot lower as most things are getting settled.

Permission to edit publication is now calculated here - https://github.com/pkp/ui-library/blob/main/src/pages/workflow/composables/useWorkflowPermissions.js#L70  . So that might be only place you need to adjust on frontend.

Let me know if you have any difficulties.

**Comment by Hafsa-Naeem** [comment]
@asmecher it needs to be reviewed

**Comment by asmecher** [comment]
Thanks, @Hafsa-Naeem! @defstat, can you review this one?

**Comment by defstat** [comment]
@Hafsa-Naeem thanks. I have added a comment [here](https://github.com/pkp/ojs/pull/4410#pullrequestreview-2577988967)  regarding that.

**Comment by Hafsa-Naeem** [comment]
@asmecher @jardakotesovec  Ready for review
**OJS_main:**
**ojs**: [pkp/pkp-lib#10263 Relax editing metadata on published/posted materials #4410](https://github.com/pkp/ojs/pull/4410)
**pkp/pkp-lib**: [pkp/pkp-lib#10263 edit metadata post publication #10335](https://github.com/pkp/pkp-lib/pull/10335)
**pkp/ui-library**: [pkp/pkp-lib#10263 Relax editing metadata on published/posted materials #398](https://github.com/pkp/ui-library/pull/398)

**Comment by asmecher** [comment]
@Hafsa-Naeem, there appear to be some test failures... Can you check them?

**Comment by ewhanson** [comment]
Only speaking on the metadata changed events I mentioned above: the current implementation looks good and covers the needed use cases. Thanks @Hafsa-Naeem!

**Comment by asmecher** [comment]
Thanks, @Hafsa-Naeem! I've added two reviews:
- https://github.com/pkp/ojs/pull/4410#pullrequestreview-2651716494
- https://github.com/pkp/ui-library/pull/398#pullrequestreview-2651723671

**Comment by Hafsa-Naeem** [comment]
@asmecher Ready for review again
**OJS_main:**
**ojs**: [pkp/pkp-lib#10263 Relax editing metadata on published/posted materials #4410](https://github.com/pkp/ojs/pull/4410)
**pkp/pkp-lib**: [pkp/pkp-lib#10263 edit metadata post publication #10335](https://github.com/pkp/pkp-lib/pull/10335)
**pkp/ui-library**: [pkp/pkp-lib#10263 Relax editing metadata on published/posted materials #398](https://github.com/pkp/ui-library/pull/398)

**Comment by asmecher** [comment]
Thanks, @Hafsa-Naeem, I've added comments to each PR!

**Comment by Hafsa-Naeem** [comment]
@asmecher Ready for review again
**OJS_main:**
**ojs**: [pkp/pkp-lib#10263 Relax editing metadata on published/posted materials #4410](https://github.com/pkp/ojs/pull/4410)
**pkp/pkp-lib**: [pkp/pkp-lib#10263 edit metadata post publication #10335](https://github.com/pkp/pkp-lib/pull/10335)
**pkp/ui-library**: [pkp/pkp-lib#10263 Relax editing metadata on published/posted materials #398](https://github.com/pkp/ui-library/pull/398)

**Comment by Hafsa-Naeem** [comment]
Thanks! @asmecher I updated the PR to address the review comments, but it looks like GitHub hasn't marked them as outdated, possibly because the changes were made within the same code blocks. Could you review the changes to confirm if the concerns have been resolved? If there are any specific comments that still need adjustments, please let me know. Meanwhile, I’m updating the PR with the latest review comment.

**Comment by asmecher** [comment]
@Hafsa-Naeem, this one still needs some work before it's ready to merge:
- There are [new review comments here](https://github.com/pkp/pkp-lib/pull/10335#pullrequestreview-2704056143),
- The PRs to are to `main`, where they should be to `stable-3_5_0` for inclusion in 3.5.0.

The changes also need to be ported to OMP and OPS before they can be merged.
I'd also suggest using consistent branch names -- e.g. `i10263-fix` rather than `edit_metadata_post_publication`, `relax_edit_metadata`, and `pkp_edit_metadata_post_publication`.

I think it's too late to get this change done in time for 3.5.0rc2, so I'd suggest @Hafsa-Naeem and @Vitaliy-1 spend a bit of time working through it to make sure it's consistent, safe, and reviewed for merge between 3.5.0rc2 and 3.5.0-0. It's better to get the policy/permission code done right rather than trying to rush it through quickly.

**Comment by Hafsa-Naeem** [comment]
Thanks @asmecher for the feedback! I’m going through the review comments to updating the permission code. along with target branches, and ensuring consistency in branch names.

**Comment by Vitaliy-1** [comment]
@Hafsa-Naeem, what is the progress on this? Is any help needed?

**Comment by Hafsa-Naeem** [comment]
@Vitaliy-1 Here’s the current status:

Ready for review:
OJS stable-3_5_0: [pkp/pkp-lib#10263 Relax editing metadata on published/posted materials #4806](https://github.com/pkp/ojs/pull/4806)
OPS  stable-3_5_0: [ pkp/pkp-lib#10263 Relax editing metadata on published/posted materials](https://github.com/pkp/ops/pull/944)

pkp-lib  stable-3_5_0: [pkp/pkp-lib#10263 Relax editing metadata on published/posted materials #11225](https://github.com/pkp/pkp-lib/pull/11225)

In progress
OMP stable-3_5_0: In progress – currently fixing tests.

**Comment by Hafsa-Naeem** [comment]
@Vitaliy-1 @asmecher 

Ready for review:
OJS (stable-3_5_0): [pkp/pkp-lib#10263 Relax editing metadata on published/posted materials #4806](https://github.com/pkp/ojs/pull/4806)

OPS (stable-3_5_0): [ pkp/pkp-lib#10263 Relax editing metadata on published/posted materials](https://github.com/pkp/ops/pull/944)

pkp-lib (stable-3_5_0): [pkp/pkp-lib#10263 Relax editing metadata on published/posted materials #11225](https://github.com/pkp/pkp-lib/pull/11225)

ui-library (stable-3_5_0): [pkp/pkp-lib#10263 Relax editing metadata on published/posted materials #581](https://github.com/pkp/ui-library/pull/581)


OMP (stable-3_5_0): [pkp/pkp-lib#10263 Relax editing metadata on published/posted materials #1943](https://github.com/pkp/omp/pull/1943)

**Comment by Hafsa-Naeem** [comment]
@Vitaliy-1 Ready for review.

**Comment by Vitaliy-1** [comment]
@Hafsa-Naeem, trying to test that. In our dataset, submission with ID 1: it has 2 publications from which 1 is published. When I'm logged as an author, I can make changes to the published publication. Nor UI or policy doesn't restrict that.

**Comment by Vitaliy-1** [comment]
For some strange reason the latest commit didn't applied cleanly. After that I see that the only problem is that the author is able to make changes to the unpublished publication through the UI but is prohibited by Policy if any publication is published. But that's a minor issue.

**Comment by Vitaliy-1** [comment]
Thanks, I've made a review. For documenting purposes, in the final version:

1. If any publication of the submission is published, we don't allow authors to make any changes to any publication of that submission, even if there are other unpublished publications.
2. The same if there is a publication with status scheduled for publication.
3. On other circumstances we allow authors to make changes after submission is submitted only when editor deliberately allows that - see canChangeMetadata flag. 
4. Admin and journal managers are always allowed to make changes to the published publication (with a warning)
5. Users assigned with other roles (editors and assistants) can change metadata of published publications if assigned and are allowed to, see canChangeMetadata flag.

**Comment by jardakotesovec** [comment]
My expectation would be that the Author can't make any change to the publication which is schleduled/published, but if there is multiple publications - I would expect that he still could change the unpublished/unschleduled ones.  ¯\_(ツ)_/¯

**Comment by Vitaliy-1** [comment]
That would be my expectations for 3.6 also, alongside with other changes. My concern here is how feasible it would be implement for 3.5 taking into account our timeframe. Currently our access policy considers publication edit access per submission basis

**Comment by Hafsa-Naeem** [comment]
@Vitaliy-1 @asmecher  Ready for review:
OJS (stable-3_5_0): [pkp/pkp-lib#10263 Relax editing metadata on published/posted materials #4806](https://github.com/pkp/ojs/pull/4806)

OPS (stable-3_5_0): [ pkp/pkp-lib#10263 Relax editing metadata on published/posted materials](https://github.com/pkp/ops/pull/944)

pkp-lib (stable-3_5_0): [pkp/pkp-lib#10263 Relax editing metadata on published/posted materials #11225](https://github.com/pkp/pkp-lib/pull/11225)

ui-library (stable-3_5_0): [pkp/pkp-lib#10263 Relax editing metadata on published/posted materials #581](https://github.com/pkp/ui-library/pull/581)


OMP (stable-3_5_0): [pkp/pkp-lib#10263 Relax editing metadata on published/posted materials #1943](https://github.com/pkp/omp/pull/1943)

**Comment by asmecher** [comment]
@Hafsa-Naeem, I requested [one small change](https://github.com/pkp/ojs/pull/4806#pullrequestreview-2807585964) that'll affect several of the repos -- they could also use a rebase to resolve submodule merge conflicts. I see there is still an incomplete conversation with @Vitaliy-1 on author permissions too; did that get resolved?

**Comment by Hafsa-Naeem** [comment]
@asmecher yes, this [conversation](https://github.com/pkp/pkp-lib/pull/11225#discussion_r2040166497) was resolved as after the meeting discussion we decided on that: if any publication of the submission is published, we don't allow authors to make any changes to any publication of that submission, even if there are other unpublished version publications [here](https://github.com/pkp/pkp-lib/issues/10263#issuecomment-2828561473). That’s why the fix ended up taking a different approach.

Let me quickly apply your suggested change and rebase to fix all the recent conflicts.

**Comment by jonasraoni** [comment]
I've added extra comments to the PRs, but the only important thing is to avoid using `===`/`!==` with non-string fields, due to a bug where the database values are retrieved as a string, until we come up with a generic fix.

**Comment by Hafsa-Naeem** [comment]
@jonasraoni Thanks for the clean-up suggestions. Since the original branch has already been merged, I’ve opened a small follow-up PR (11386)](https://github.com/pkp/pkp-lib/pull/11386) to address the feedback

**Comment by Vitaliy-1** [comment]
@Hafsa-Naeem, thanks, reviewed! It requires minor tweaks before merging.

**Comment by asmecher** [comment]
@Vitaliy-1 and @Hafsa-Naeem, this one is still a blocker for 3.5.0-0, and needs some time for testing before we can get the release out. It looks like it's nearly ready, but please consider it a high priority!

**Comment by Hafsa-Naeem** [comment]
@asmecher This has now been merged to stable-3_5_0.

**Comment by asmecher** [comment]
Excellent, thanks, @Hafsa-Naeem! Please don't forget the forward-port to `main`, if you haven't done that already.

@Tribunal33, this is ready for some testing as a high priority!

**Comment by Tribunal33** [comment]
This passes QA.  Couldn't find issues with it but hard to tell as there are no requirements.  Even added the URN plugin to throw it for a loop.  Seems solid.  Here is the testing summary.

Test scenarios for OJS:
1. As Editor, can make changes to submission in the published stage
2. As Editor, the red warning message banner reads "Warning: This version has been published. Editing it may impact the published content."
3. When changes are made and then Save button is selected no additional warning message appears and a green check mark appears by the button showing the changes have been saved.
4. As Editor, all fields including language translations in the Title & Abstract can be altered and saved
5. As Editor, Unpublish and Create new Version buttons still function normally
6. As Editor, Top level action button View can see the published result changes.
7. As Editor, Top Level action button Activity Log will display the Date, User and Event "Submission metadata updated"
8. As Editor all options to modify Contributors can be saved
9. As Editor, all Metadata keywords can still be added and saved
10. As Editor, All JATS XML functionality works and previous saved changes are reflected in the generated JATS XML printout
a. Can check for keyword updates done on previous test by using ctrl-f and finding the keyword.
11. As Editor, All Galleys functionality works (can add a galley, view, change file, and delete)
12.  As Editor, all fields from Permissions & Disclosure can be updated and saved
13. As Editor, all fields from Issue can be updated and saved and should be visible with View button (Journal display)
14. As Editor, Adding new and clearing old URN Identifiers works (plugin needs to be installed).
15. As Author it is locked down with the Save button being disabled and changes are not saved even though fields can be altered
16. As Author the red warning message banner reads "This version has been published and can not be edited."
17. Repeat all for OMP excluding those that don't apply
18. Repeat all for OPS excluding those that don't apply

**Comment by asmecher** [comment]
@Hafsa-Naeem Is this one ready to be closed?

**Comment by Vitaliy-1** [comment]
Requires porting to main

**Comment by Hafsa-Naeem** [comment]
@asmecher @Vitaliy-1  Forward ported to `main` ready for review and merge.

**OJS main**: [Relax editing metadata on published/posted materials #4902](https://github.com/pkp/ojs/pull/4902)

**pkp/pkp-lib**: [pkp/pkp-lib#10263 Relax editing metadata on published/posted materials #11431](https://github.com/pkp/pkp-lib/pull/11431)

**pkp/ui-library**: [pkp/pkp-lib#10263 Relax editing metadata on published/posted materials #634](https://github.com/pkp/ui-library/pull/634)

**omp main**: [pkp/pkp-lib#10263 Relax editing metadata on published/posted materials #2006](https://github.com/pkp/omp/pull/2006)

**ops main**: [pkp/pkp-lib#10263 Relax editing metadata on published/posted materials #990](https://github.com/pkp/ops/pull/990)

**Comment by asmecher** [comment]
@Vitaliy-1, what's the status of this one? It looks like it was merged to `stable-3_5_0`, but not `main`, and is waiting for your review [on the PRs listed](https://github.com/pkp/pkp-lib/issues/10263#issuecomment-2914608025).

**Comment by Vitaliy-1** [comment]
@asmecher, yes, you are right, giving it a lower priority among other PRs and tasks.

**Comment by Vitaliy-1** [comment]
Thanks, @Hafsa-Naeem. I left minor comments but they aren't showstoppers

**Comment by asmecher** [comment]
@Hafsa-Naeem, just a reminder that this one is waiting for your next steps, I think.

**Comment by Hafsa-Naeem** [comment]
@Vitaliy-1 I've replied to the comments, please let me know if I missed anything. so then I can rebase and we can proceed with the merge.

**Comment by asmecher** [comment]
@Hafsa-Naeem, I found a case in OPS `stable-3_5_0` where the API seems to be rejecting changes when they should be allowed after this change... See the screencap.

https://github.com/user-attachments/assets/4b43698b-dc2d-4e29-aa9c-0567e06930cc

**Comment by Vitaliy-1** [comment]
Latest changes look ok but I see that tests on OJS are failing. @Hafsa-Naeem, let me know if anything here requires my attention.

**Comment by Hafsa-Naeem** [comment]
@asmecher @Vitaliy-1  Ready for review:
OJS (stable-3_5_0): [pkp/pkp-lib#10263 Allow journal editors to edit publications without participant assignment #5061](https://github.com/pkp/ojs/pull/5061)
pkp-lib (stable-3_5_0): [pkp/pkp-lib#10263 Allow journal editors to edit publications without participant assignment #11729](https://github.com/pkp/pkp-lib/pull/11729)
OPS (stable-3_5_0): [pkp/pkp-lib#10263 Submodule update ##Hafsa-Naeem/i10263-stable_3_5_0-follow-up #1085](https://github.com/pkp/ops/pull/1085)
OMP (stable-3_5_0): [pkp/pkp-lib#10263 Submodule update ##Hafsa-Naeem/i10263-stable_3_5_0-follow-up #2103](https://github.com/pkp/omp/pull/2103)

**Comment by asmecher** [comment]
@Hafsa-Naeem, on [the PRs above](https://github.com/pkp/pkp-lib/issues/10263#issuecomment-3218224961) to resolve [the problem with editing relation data](https://github.com/pkp/pkp-lib/issues/10263#issuecomment-3212476570), it looks like the PRs just work around the fact that `canEditPublication` is returning `false` on published content. I think the function itself needs to be updated, so that it accurately reports whether a user is able to edit the submission metadata or not.

**Comment by Hafsa-Naeem** [comment]
@asmecher   Ready for review:
OJS (stable-3_5_0): [pkp/pkp-lib#10263 Allow journal editors to edit publications without participant assignment #5061](https://github.com/pkp/ojs/pull/5061)
pkp-lib (stable-3_5_0): [pkp/pkp-lib#10263 Allow journal editors to edit publications without participant assignment #11729](https://github.com/pkp/pkp-lib/pull/11729)
OPS (stable-3_5_0): [pkp/pkp-lib#10263 Submodule update ##Hafsa-Naeem/i10263-stable_3_5_0-follow-up #1085](https://github.com/pkp/ops/pull/1085)
OMP (stable-3_5_0): [pkp/pkp-lib#10263 Submodule update ##Hafsa-Naeem/i10263-stable_3_5_0-follow-up #2103](https://github.com/pkp/omp/pull/2103)

**Comment by Hafsa-Naeem** [comment]
@asmecher  Ready for review:
OJS (stable-3_5_0): [pkp/pkp-lib#10263 Submodule update ##Hafsa-Naeem/i10263-stable_3_5_0-follow-up#5061](https://github.com/pkp/ojs/pull/5061)
pkp-lib (stable-3_5_0): [pkp/pkp-lib#10263 Allow journal editors to edit publications without participant assignment #11729](https://github.com/pkp/pkp-lib/pull/11729)
OPS (stable-3_5_0): [pkp/pkp-lib#10263 Submodule update ##Hafsa-Naeem/i10263-stable_3_5_0-follow-up #1085](https://github.com/pkp/ops/pull/1085)
OMP (stable-3_5_0): [pkp/pkp-lib#10263 Fix submission mapper crash when review assignments are null#2103](https://github.com/pkp/omp/pull/2103)

**Comment by asmecher** [comment]
@Hafsa-Naeem, looks good, thanks -- just a question [on the OMP PR](https://github.com/pkp/omp/pull/2103#discussion_r2304952921).

**Comment by Hafsa-Naeem** [comment]
Thanks @asmecher. I’ll port it to main now.

**Comment by kaitlinnewson** [comment]
Hi all, will this be ported to main soon? I'd like to ensure things are working properly for the export plugins alongside the new JAV changes (e.g. that items are marked as needing to be synced for export when metadata is changed). Thanks!

**Comment by Hafsa-Naeem** [comment]
@asmecher @Vitaliy-1 could you please approve these PRs so I would rebase and merge them. They require two approvals. Thanks
OJS (stable-3_5_0): [pkp/pkp-lib#10263 Submodule update ##Hafsa-Naeem/i10263-stable_3_5_0-follow-up#5061](https://github.com/pkp/ojs/pull/5061)
pkp-lib (stable-3_5_0): [pkp/pkp-lib#10263 Allow journal editors to edit publications without participant assignment #11729](https://github.com/pkp/pkp-lib/pull/11729)
OPS (stable-3_5_0): [pkp/pkp-lib#10263 Submodule update ##Hafsa-Naeem/i10263-stable_3_5_0-follow-up #1085](https://github.com/pkp/ops/pull/1085)
OMP (stable-3_5_0): [pkp/pkp-lib#10263 Fix submission mapper crash when review assignments are null#2103](https://github.com/pkp/omp/pull/2103)

**Comment by asmecher** [comment]
@Hafsa-Naeem Approved, thanks!

**Comment by Vitaliy-1** [comment]
@Hafsa-Naeem, thanks, look good and can be merged after rebase

**Comment by asmecher** [comment]
@Hafsa-Naeem, is the port to `main` still in progress?

**Comment by Hafsa-Naeem** [comment]
@asmecher Yes, almost done with the port to main. A few tests are failing due to frontend changes... I’m working through those and will push the updates once all tests pass.

**OJS main**: Merged - [Relax editing metadata on published/posted materials #4902](https://github.com/pkp/ojs/pull/4902) 
**pkp/pkp-lib**: Merged - [pkp/pkp-lib#10263 Relax editing metadata on published/posted materials #11431](https://github.com/pkp/pkp-lib/pull/11431)
**pkp/ui-library**:  Merged -  [pkp/pkp-lib#10263 Relax editing metadata on published/posted materials #634](https://github.com/pkp/ui-library/pull/634)
**ops main**:  Merged - [pkp/pkp-lib#10263 Relax editing metadata on published/posted materials #990](https://github.com/pkp/ops/pull/990)

**omp main**: Fixing tests -  [pkp/pkp-lib#10263 Relax editing metadata on published/posted materials #2006](https://github.com/pkp/omp/pull/2006)

**Comment by asmecher** [comment]
@Hafsa-Naeem, I noticed that this broke testing in OMP main, which I fixed with [this commit](https://github.com/pkp/omp/commit/bf660104b55b58738c00deea942993534e4afad1). However, I see that the old `publication.editDisabled` locale key is still in the codebase in a number of places; I think it should be removed.

**Comment by Hafsa-Naeem** [comment]
@asmecher we still use `publication.editDisabled` locale key in the author workflow to prevent them from editing metadata after it's published. Let me know if you think it should be renamed or scoped differently. Thanks.

**Comment by asmecher** [comment]
@Hafsa-Naeem Ah, that makes sense.

Is there still outstanding work on this issue?

**Comment by Hafsa-Naeem** [comment]
@asmecher Currenlty running tests on github. Once they pass it's ready to merge.
update: Some of the Cypress data suite tests are failing due to recent UI changes... mostly selector and metadata assertion mismatches. These tests rely heavily on frontend components that have recently shifted. I'm updating them now and will re-run CI and merge once everything passes. I applied a similar set of updates in OJS/OPS before merging them recently, so this should be straightforward.

**Comment by Hafsa-Naeem** [comment]
@asmecher @Vitaliy-1 This follow-up allows editors to add/edit contributors after a publication is published. The original PR addressed metadata editing but in `main` the `addContributor()` was still blocked by `api.publication.403.cantEditPublished`.

Follow-up PRs:
pkp-lib(main): [pkp/pkp-lib#10263 fix to allow editors to add/edit contributors on published publications #12075](https://github.com/pkp/pkp-lib/pull/12075)

ojs(main): [pkp/pkp-lib#10263 Submodule update ##Hafsa-Naeem/i10263-main-follow-up#5201](https://github.com/pkp/ojs/pull/5201)

ops(main): [pkp/pkp-lib#10263 Submodule update ##Hafsa-Naeem/i10263-main-follow-up #1146
](https://github.com/pkp/ops/pull/1146)

omp(main): [pkp/pkp-lib#10263 Submodule update ##Hafsa-Naeem/i10263-main-follow-up #1146
](https://github.com/pkp/omp/pull/2178)

**Comment by asmecher** [comment]
@Hafsa-Naeem, I added [a comment on the `pkp-lib` PR](https://github.com/pkp/pkp-lib/pull/12075/files#r2578906452). Is the problem that the `contributorRolesToParams` function is getting called with a `roleIds` parameter that contains objects, not numeric IDs? If so, then I think it's better to solve the mistake in the calling code, rather than making the `contributorRolesToParams` function behave differently.

**Comment by Tribunal33** [comment]
@Hafsa-Naeem please let me know if these additional changes need to be tested.

**Comment by Hafsa-Naeem** [comment]
@asmecher @Vitaliy-1 Ready for review
Follow-up PRs:
pkp-lib(main): [pkp/pkp-lib#10263 fix to allow editors to add/edit contributors on published publications #12075](https://github.com/pkp/pkp-lib/pull/12075)

ojs(main): [pkp/pkp-lib#10263 Submodule update ##Hafsa-Naeem/i10263-main-follow-up#5201](https://github.com/pkp/ojs/pull/5201)

**Comment by asmecher** [comment]
@Hafsa-Naeem, I'm still not sure where all the ambiguity over `$params['contributorRoles']` is coming from -- it looks like you've coded it to expect an array of scalars (ints or numeric strings), or arrays containing an `id` or `roleId` index. Where is this ambiguity coming from? Do you actually see data coming into the endpoint in all these forms?

**Comment by Hafsa-Naeem** [comment]
@asmecher @Vitaliy-1 Ready for review
pkp-lib(main): [pkp/pkp-lib#10263 fix to allow editors to add/edit contributors on published publications #12075](https://github.com/pkp/pkp-lib/pull/12075)

ojs(main): [pkp/pkp-lib#10263 Submodule update ##Hafsa-Naeem/i10263-main-follow-up#5201](https://github.com/pkp/ojs/pull/5201)

**Comment by asmecher** [comment]
@Hafsa-Naeem, I've added a comment on [the pkp-lib PR](https://github.com/pkp/pkp-lib/pull/12075#pullrequestreview-3636951058), thanks!

**Comment by Tribunal33** [comment]
@asmecher, @Vitaliy-1 or @Hafsa-Naeem.  As this ticket is still open, I was wondering if it is expected that any metadata (even through new plugins) should be able to be modified after publishing?  #12071 , for example, has a new funding field.  I added a test to make sure it worked with published articles as this is relaxed metadata is in the back of my mind.  Going forward, however, will any addition to the publication space be automatically given this relaxed metadata practice or will each new addition need to be manually added?

**Comment by Hafsa-Naeem** [comment]
@asmecher @Vitaliy-1  Ready for another review.


I've removed the extra normalisation and now pass the array straight through to contributorRolesToParams() as before, keeping this change minimal and in line with the existing pattern. This should address the concern regarding the unnecessary checks and ambiguity. 


pkp-lib(main): [pkp/pkp-lib#10263 fix to allow editors to add/edit contributors on published publications #12075](https://github.com/pkp/pkp-lib/pull/12075)

ojs(main): [pkp/pkp-lib#10263 Submodule update ##Hafsa-Naeem/i10263-main-follow-up#5201](https://github.com/pkp/ojs/pull/5201)

**Comment by asmecher** [comment]
@Hafsa-Naeem, I've added [one more comment](https://github.com/pkp/pkp-lib/pull/12075/files#r2683853929) to the pkp-lib PR.

**Comment by Hafsa-Naeem** [comment]
@asmecher  Ready for review.. I've removed the unused `api.publication.403.cantEditPublished` key from the locale files.


pkp-lib(main): [pkp/pkp-lib#10263 fix to allow editors to add/edit contributors on published publications #12075](https://github.com/pkp/pkp-lib/pull/12075)

ojs(main): [pkp/pkp-lib#10263 Submodule update ##Hafsa-Naeem/i10263-main-follow-up#5201](https://github.com/pkp/ojs/pull/5201)

**Comment by asmecher** [comment]
@Tribunal33, merged ready for your testing. The latest changes should permit contributors to be edited (by privileged users) on already-published content.

**Comment by Tribunal33** [comment]
@asmecher , @Hafsa-Naeem . This fails QA.  Previously it was tested to see if contributors could be edited/added and it worked.  Not sure if there is a new piece that I should test.  I did noticed one thing though.  I tested the changes on an older version to compare and didn't immediately see a difference.  They could be added and changed.  However I noticed that with most recent change the saved changes are no updating the article landing page correctly.

<img width="2073" height="666" alt="Image" src="https://github.com/user-attachments/assets/84f5a4c1-95bf-48c2-8b2c-57df8cb9604e" />

**Comment by Hafsa-Naeem** [comment]
@Tribunal33  Could you please confirm whether the contributor edits you expected to see on the article landing page were made on the published publication version? 
About the old version where the test passed, that publication only had one version and it was already published. In the current UI/backend, before this fix, as Bozana reported in Mattermost [("addContributor() could still be blocked on published publications because it returned api.publication.403.cantEditPublished)](https://mattermost.publicknowledgeproject.org/pkp/pl/18n9ybgzdpbq5ktce7amo555fr), So the older behaviour wasn’t fully aligned with the current UI expectations.

I tried to reproduce the QA failure issue locally using the same submission. This article has two publication versions. one published ("Version of Record 1.0") and one unpublished ("Version of Record 1.1"). Since the landing page is always rendered from the latest published version, the behaviour depends on which version is edited. 
When I update contributors on the published ("Version of Record 1.0"), the landing page reflects the changes correctly (name edits and newly added anonymous contributor). I've attached screenshots showing this.
But when I update contributors on the unpublished ("Version of Record 1.1"), the landing page does not change, which is expected, for unpublished versions. 
If possible, could you double-check which publication version was edited during your test? If the edits were made on the published version and still didn’t appear, please let me know the exact steps and I'll reproduce and fix it right away.
Thanks for the detailed testing! Let me know what you find.

<img width="1786" height="712" alt="Image" src="https://github.com/user-attachments/assets/92f43eb0-9863-465a-8878-e0dc9ce61713" />

<img width="946" height="917" alt="Image" src="https://github.com/user-attachments/assets/894f2378-174a-4870-835e-ff69acefea8e" />

<img width="1766" height="646" alt="Image" src="https://github.com/user-attachments/assets/e82943d9-caa1-4852-bd67-e5b365248f81" />

<img width="1103" height="1001" alt="Image" src="https://github.com/user-attachments/assets/c0e0e8e1-ad5f-403a-9b16-83bbe5f790ac" />

**Comment by Tribunal33** [comment]
@Hafsa-Naeem sorry that was my mistake.  It was a published version that was published and then unpublished.  This works as expected.  After re-publishing all changes were updated.  Swapping between versions on the article landing page worked too.   I added a test to this to check for multiple version changes with contributor as well.
